source/funwithflags/use_cases/signup.py

```diff
@@ -0,0 +1,26 @@
+"""Module for user signup implementation."""
+from funwithflags.definitions import (
+    SignupRequest,
+    User
+)
+from funwithflags.entities import generate_salt, hash_password
+from funwithflags.gateways import Context
+
+
+def signup(request: SignupRequest, context: Context):
+    """Given a `SignupRequest` object and `Context` of gateways, execute signup logic.
+    This will create a valid `User` object and write it into Postgres database table.
+    Return user id got from db gateway.
+    """
+    salt: bytearray = generate_salt()
+    hashed_password: bytearray = hash_password(request.password, salt)
```

Comment:
Consider use `bytes` instead of `bytearray`. There aren't use cases where we need it to be mutable.

Reply:
Done.